core/src/main/java/org/infinispan/interceptors/xsite/BaseBackupInterceptor.java

```diff
@@ -55,33 +53,30 @@ public Object visitPrepareCommand(TxInvocationContext ctx, PrepareCommand comman
          return invokeNext(ctx, command);
       }
 
-      BackupResponse backupResponse = backupSender.backupPrepare(command, ctx.getCacheTransaction());
-      return processBackupResponse(ctx, command, backupResponse);
+      InvocationStage stage = backupSender.backupPrepare(command, ctx.getCacheTransaction(), ctx.getTransaction());
+      return invokeNextAndWaitForCrossSite(ctx, command, stage);
```

Comment:
I don't see the issue TBH. 
Before was send-xsite, send-local-cluster, wait-xsite and now it is the same. 
If a local cluster throws an Exception, the transaction is aborted and the rollback is sent. 

The Transaction is needed because it is our public `CustomFailurePolicy`. I can inject the TransactionTable and get the Tx from there... but it is one more lookup to do :/

Reply:
The difference is that `BackupResponseHandler.onResponse()` is now called in parallel with `invokeNext()`, before the backup responses used to be processed in `thenAccept()`.
It's probably ok now because the things the backup response handler needs do not change during invocation, but I don't like special cases like this, and the `Transaction` is a wildcard because the javadoc doesn't say anything about thread safety.

BTW, I created ISPN-10683 to remove CustomFailurePolicy transaction parameters

Reply:
I think it is fine. Although the `CustomFailurePolicy` has access to the cache, I don't believe it can "mess up" the command invocation in any way. 
About the `Transaction` I see it as an identifier of the failing transaction. I doubt the user will interact with it in any way.

Reply:
Ok, I'm still worried that if it breaks in the future it will be really hard to track down the cause, but it's not a deal-breaker.